src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+from .. import Dataset
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = Dataset()
+    subdomain.meta = dataset.meta.copy()
+
+    for name, group in dataset.items():
+        pos = group.get("position", group.parent["amr"]["position"])
+        if pos.shape[0] != group.shape:
+            continue
```

Comment:
Interesting question: What should we do if the group has no position and is not AMR data? Should we emit a warning? Should we silently skip the group? Should we include the entire group? (e.g. if the user added a custom group of metadata)

Reply:
I'm down for a simple warning, just so users don't get confused if the subdomain is "missing" a `Datagroup`.

Reply:
Something like this perhaps?
```
warnings.warn(
     "Ignoring datagroup '{}', which has no position " +
     "vector and has different shape than 'amr' group.".format(group))
```